### ISSUE TITLE
Cache storage documentation

### ISSUE THREAD

**Issue by mfulton26** [apache/cordova-docs#1210]
# Feature Request

Details on advantages/disadvantages, supported platforms, etc. for the Cache API supported by browsers, etc.

## Motivation Behind Feature

Cordova has some great docs for helping when making decisions on where and how to store data but doesn't currently provide information saved about the Cache API which can be used in some cases instead of IndexedDB. Documenting this will help folks make informed decisions.

## Feature Description

Document Cache API similar to how localStorage, IndexedDb, Web SQL, etc. are documented.

www/docs/en/dev/cordova/storage/storage.md

## Alternatives or Workarounds

Various sites out there have some information about total size limits on iOS, etc. etc. but it is hard to find what applies to apps running inside Cordova/WKWebView.

**Comment by breautek** [comment]
Based on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Cache), it looks like the Cache API is a service worker feature. I'm pretty sure service workers are **not** supported by the underlying webviews. They are supported in the browser application themselves (e.g. Safari and Chrome) but not the WKWebView / Android System Webview counterpart in which the native apps have access to.

**Comment by mfulton26** [comment]
Yes, if it could be tested and documented that would be very helpful for reference. Especially for iOS.

**Comment by Shiva903-hue** [comment]
Hi, I’d like to work on this issue and contribute documentation around Cache Storage usage in Cordova.

I plan to add a clear, docs-only section explaining:

How Cache Storage is exposed in Cordova WebView (window.caches)

Platform differences and expectations (Android WebView vs iOS WKWebView)

Persistence and eviction caveats

When Cache Storage is appropriate vs when alternatives like IndexedDB should be used

I’ll avoid making assumptions about exact storage limits and will focus on documented behavior and practical guidance.

If this approach looks good, I’m happy to take ownership of the issue and open a PR for it.
Feel free to assign it to me so others don’t duplicate the work.

**Comment by breautek** [comment]
@Shiva903-hue We can probably expand on [storage page](https://cordova.apache.org/docs/en/dev/cordova/storage/storage.html) (see https://github.com/apache/cordova-docs/blob/master/www/docs/en/dev/cordova/storage/storage.md) for caches but we should try to refrain from documenting the feature because it's a web feature. We can link to resources such as MDN that can document the web feature which will be much more complete and more likely to be updated as web standards change. We can however summarize the web feature and how they work in conjunction in webviews, especially if they exhibit caveats within a webview vs a typical browser context.

It's not a feature I actually ever use, at least not directly, and I'm not very knowledgeable about this particular API. I'd have to trust the community input for accuracy.

I assume they exhibit similar issues as other web storage contexts where they are webview and origin dependent (so changing the origin like setting a different scheme, or changing webview implementations will prevent access to the original cache object, if they persist across app restarts).

> Platform differences and expectations (Android WebView vs iOS WKWebView)

If possible, I'd throw Electron in the mix too, for completeness, but not required. Android and iOS platforms are definitely the most important platforms to cover imo.

> I’ll avoid making assumptions about exact storage limits and will focus on documented behavior and practical guidance.

100% 👍

**Comment by Shiva903-hue** [comment]
@breautek 
Thanks for catching that. I have moved the Cache Storage section to sit after IndexedDB and before Plugin-Based Options as suggested.
